serve/mlc_serve/model/tvm_model.py

```diff
@@ -203,6 +204,16 @@ def profile_memory_usage(self, seq_lens):
 
         return self.get_used_memory()
 
+    def get_logprob_infos(
+        self,
+        i: int,
+        logprob_infos: Optional[RawLogprobsInfos],
+    ) -> Optional[RawLogprobsInfos]:
+        if logprob_infos is None or logprob_infos[i] is None:
+            return None
+        return [logprob_infos[i]]
```

Comment:
No need for this to be a method of this class. Move it to `model_common.py`.

Reply:
Sorry, but I also can not do it. Earlier it was done so, but led to performance reduction. In common case we have two options `logprob_infos` is 1. `None`, case when logprobs not requested, such trick allows to avoid performance reduction for this case. 2. list of `None` or `logprob_info`.
When we create `TextGenerationResult` we need only one element from the list. It means we should check that it is a list (not None) and extract the element, it can not be transfer to `model_common.py` side.
My doubts are we potentially can expect one element of `RawLogprobsInfo` in `TextGenerationResult` instead of the list with one element, it will slightly simplified the code. But I've seen that `generated_tokens` is [a list with some comments about speculative decoding](https://github.com/octoml/mlc-llm/blob/2b3fcf0ab666258c6add5dfff8676c05e013fcc8/serve/mlc_serve/engine/model_module.py#L50C1-L52C32). And I've decided that logprobs info should correspond to each token in the list (i.e. should be the list)

Reply:
I see what you mean. This function doesn't touch `self` at all, but its implementation is valid only when used by this class. So it cannot be moved to `model_common.py` which is supposed to be a collection of general utilities. 

But soon I'm adding a PyTorch-based implementation of the model, in a file `pt_model.py`. I don't want to repeat this class there. Assuming that `get_logprob_infos` is called only by the TVM or by the PT model, I think it is ok to put it inside `model_common.py`.

Reply:
https://github.com/octoml/mlc-llm/pull/184